README.md

```diff
@@ -375,6 +388,267 @@ whitout changes, the effect will be run. (This function does not return anything
     };
 ```
 
+## Hocs
+ 
+### `enhanceWithClickOutside`
+This **[HOC][11]** allows to the wrapped component to call a function when a click event occurs outside the component. The function that will be call must be named handleClickOutside and should receive an event as parameter.
+
+### Example
+
+```javascript
+
+class ToBeEnhanced extends React.Component<any> {
+
+    constructor(props) {
+        super(props);
+    }
+
+    public render() {
+
+        return (
+            <div
+                style={{
+                    backgroundColor: 'red',
+                    height: '100px',
+                    width: '100px',
+                }}
+            />
+        );
+    }
+
+    private handleClickOutside = (e) => {
+        //Do what you want in here.
+    }
+}
+
+const Enhanced = enhanceWithClickOutside(ToBeEnhanced);
+
+const Test = () => {
+
+ return (
+  <Enhanced />
+ );
+};
+
+```
+
+
+## Functions
+ 
+### `colorGenerator`
+This function retrieves an **[Array][1]** of RGB colors that are a result of an interpolation between to given RGB colors.
+
+### Parameters
+- `startColor` **[Array][1]** RGB color (<Red|Green|Value>) that represents the start color for the interpolation.
+- `endColor` **[Array][1]** RGB color (<Red|Green|Value>) that represents the end color for the interpolation.
+- `factor` **[Number][5]** The ratio of the output rate to the input rate(startColor -> endColor).
+
+### Returns
+- `Colors` **[Array][1]** RGB colors array.
+
+### Example
+
+```javascript
+
+const startColor = [255, 0, 0];
+const endColor = [0, 255, 0];
+
+const colors = ColorGenerator(startColor, endColor, 1);
+
+```
+
+### `debounce`
+ This function allow to apply a function after certain amount of time.
+### Parameters
+- `F` **[function][2]** A function to be executed.
+- `waitMilliseconds ` **[number][2]** The number of miliseconds to wait before execute the function.
+
+### Example
+
+```javascript
+const myFormComponent = () => {
+
+ const [text, setText] = React.useState("");
+ 
+ const debounceOnClick = () =>
+  debounce(()=>setText("onClick"), 200);
+ 
+ return(
+  <div>
+    <h1>{text}</h1>
+    <button onClick={debounceOnClick}> Reset </button> 
+  <div>
+ );
+};
+```
+
+### `requestController`
+This function allows to create request to an specific url and handle the serve response.
```

Comment:
Just document createFetchController and try to focus on the public API of this function.